src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -78,6 +79,41 @@ export async function setFirstDataSourceAsDefault(
   }
 }
 
+export function getFilterDataSources(
+  dataSources: Array<SavedObject<DataSourceAttributes>>,
+  filter?: (dataSource: SavedObject<DataSourceAttributes>) => boolean
+) {
+  return filter ? dataSources.filter((ds) => filter!(ds)) : dataSources;
+}
+
+export function getDefaultDataSource(
+  dataSources: Array<SavedObject<DataSourceAttributes>>,
+  LocalCluster: DataSourceOption,
+  uiSettings?: IUiSettingsClient,
+  hideLocalCluster?: boolean,
+  defaultOption?: DataSourceOption[]
+) {
+  let defaultShowingDataSource: DataSourceOption;
+
+  if (defaultOption && dataSources.find((dataSource) => dataSource.id === defaultOption?.[0].id)) {
+    defaultShowingDataSource = defaultOption?.[0];
+  } else if (
+    uiSettings &&
+    uiSettings?.get('defaultDataSource', null) !== null &&
+    dataSources.find((dataSource) => dataSource.id === uiSettings?.get('defaultDataSource'))
```

Comment:
There are two locations were used `uiSettings?.get('defaultDataSource')`, is it possible to use `const defaultDataSourceId = uiSettings?.get('defaultDataSource');` to smi

Reply:
+1